File: 2026/aed-1/cormen/2.1/insertionSort.java
public class insertionSort {
    public static void sort(int[] A, int n) {
        for (int i = 1; i < n; i++) {
            int key = A[i];
            int j = i - 1;

            while (j >= 0 && A[j] > key) {
                A[j + 1] = A[j];
                j = j - 1;
            }

            A[j + 1] = key;
        }
    }
    public static void main(String[] args) {
        int A[] = {4, 3, 1, 6, 10};
        int n = A.length;

        IO.println("Unsorted.");

        for (int i = 0; i < n; i++) {
            IO.println(A[i]);
        }

        sort(A, n);

        IO.println("Sorted.");

        for (int i = 0; i < n; i++) {
            IO.println(A[i]);
        }
    }
}

File: 2026/aed-1/cormen/2.1/insertion_sort.py
#!/usr/bin/env python3


def insertion_sort(A: list[int], n: int) -> None:
    for i in range(n):
        key: int = A[i]
        j: int = i - 1

        while j >= 0 and A[j] > key:
            A[j + 1] = A[j]
            j = j - 1

        A[j + 1] = key


def main() -> None:
    A: list[int] = [4, 3, 1, 6, 10]
    n: int = len(A)

    print("Unsorted.")

    for i in range(n):
        print(A[i])

    insertion_sort(A, n)

    print("Sorted.")

    for i in range(n):
        print(A[i])


if __name__ == "__main__":
    main()
